### ISSUE TITLE
files_external: limited number of connections

### ISSUE THREAD

**Issue by kaffeeringe** [nextcloud/server#36422]
I use Nextcloud 25.0.2 with the files_external app and a Hetzner Storage-Box that I connected through WebDAV. Everything worked fine until recently and I don't know what changed. But Hetzner support tells me that the problem is, that the connections are limited to 10 and that it's doesn't work, when I exceed the 10 connections. As Nextcloud is the only software accesses the Storage-Box, there should be some way to limit the number of connections, where ever they come from: desktop client, mobile client or web UI.

**Comment by thomasmerz** [comment]
Have you tried SFTP with your storagebox? I've "connected" my storagebox this way - works pretty well 👍🏻

**Comment by kaffeeringe** [comment]
Thank you for this workaround. I'll try that. I only read that webdav was the fasted of the protocols. And basically Nextcloud should also work with webdav if that is an option in the settings.

**Comment by kaffeeringe** [comment]
It seems that the connection limit is for example a problem, when I try to delete lots of folders and files.

I uploaded ~45000 images in year/month/day folders. When I mark them all and delete them I get errors on all of them. When I try to delete a folder with only a few images in it, deleting works. When there are more images it even fails for one year folder.

**Comment by thomasmerz** [comment]
>  Thank you for this workaround. I'll try that. I only read that webdav was the fasted of the protocols. And basically Nextcloud should also work with webdav if that is an option in the settings.

I agree that NC should be fixed regarding WebDAV! But for me SFTP works fine. And I didn't find a meaningful and trustworthy and universally valid statement which is "faster" (or performs better) regarding WebDAV, CIFS, SSHFS/SFTP or NFS… 🤷🏼‍♂️ 


> It seems that the connection limit is for example a problem, when I try to delete lots of folders and files.

It seems that WebDAV uses a single connection for each file/dir operation and SFTP might not work like this - or I never have deleted thousands of files at once 😆  so I should test this with nearly 100k files on my SFTP-mounted external-Storagebox:
```
for i in {1..99}; do md dir-$i; cd dir-$i; for f in {1..999}; do touch file-$f; done; cd ..; done
```
Well, I stopped at 50k files/dirs:
```
fd|wc -l
52096
```

And voila! Delete them all:
<img width="405" alt="grafik" src="https://user-images.githubusercontent.com/18568381/215787770-472869ed-fdde-4285-9ef2-18c0a466c894.png">
And working…
<img width="989" alt="grafik" src="https://user-images.githubusercontent.com/18568381/215788364-da3d8910-498c-47d5-850a-5ccb253c56c9.png">

Well it's slow…

And oha!
<img width="269" alt="grafik" src="https://user-images.githubusercontent.com/18568381/215789631-a458ee45-fe13-4153-9532-5344c273080c.png">

And now it's not very responsive anymore… (504 Gateway Time-out) 😢  But deletion continues in the background. I should NEVER do this again 🤣

**Comment by thomasmerz** [comment]
Did you see any error message? In which location / logfile? I don't see anything unless getting this "504" for some minutes until:
<img width="676" alt="grafik" src="https://user-images.githubusercontent.com/18568381/215792810-501429b7-ce8e-443a-844d-da0c6f39e667.png">

**Comment by kaffeeringe** [comment]
I set loglevel to 2 but I don't see any entries.

**Comment by kaffeeringe** [comment]
I tried to scan all my files using occ. That failed with an error because files were locked.

I unlocked the 100k+ files. Now it's scanning again. 

Maybe that is connected somehow?

**Comment by thomasmerz** [comment]
I saw also correlated many (ten) thousands of dns queries from my NC to my Pi-hole asking for my Hetzner Storagebox 😞 So I think that NC is working one each file/dir and this seems to be very, very inefficient…

**Comment by kaffeeringe** [comment]
I tried to log directly via ssh and that doesn't fail, but rm -r * isn't too fast either. So it may not only be a problem of Nextcloud. But it would be good if Nextcloud not just failed with a generic error message.

**Comment by thomasmerz** [comment]
Another try - directly on `sshfs` mounted Storagebox (in a dir that's not visible to my NC):
```
for i in {1..50}; do md dir-$i; cd dir-$i; for f in {1..999}; do touch file-$f; done; cd ..; done
```

`rm -rf *` of these 50k dirs/files took 162s:

```
real	2m42.053s
user	0m0.203s
sys	0m2.766s
```
